infra/terraform/modules/data_access/memberships/memberships.py

```diff
@@ -114,7 +115,19 @@ def policy_arn(team_slug, policy_type):
 
 
 def bucket_name(team_slug):
-    return "{env}-{team_slug}".format(
-        env=os.environ["STAGE"],
-        team_slug=team_slug.lower()
-    )
+    '''
+    Generate the S3 bucket name by prefixing the environment name and
+    replacing invalid characters with an hyphen ('-').
+
+    NOTE: This is a very simple implementation which doesn't cover all the
+          S3 limitations (e.g. max length or labels limitations, etc...)
+
+    See: http://docs.aws.amazon.com/en_gb/AmazonS3/latest/dev/BucketRestrictions.html
+    '''
+
+    INVALID_BUCKET_CHARS = r"[^a-z0-9.-]"
+
+    name = team_slug.lower()
+    name = re.sub(INVALID_BUCKET_CHARS, "-", name)
```

Comment:
This code should cover at least the following S3 bucket naming rules:

* Bucket names must be a series of one or more labels. Adjacent labels are separated by a single period (.).
* Bucket names can contain lowercase letters, numbers, and hyphens. Each label must start and end with a lowercase letter or a number.
* Bucket names must not be formatted as an IP address (e.g., 192.168.5.4).